projects/kit/pipes/filter-by-input/filter-by-input.pipe.ts

```diff
@@ -29,6 +32,11 @@ export class TuiFilterByInputPipe
         items: ReadonlyArray<readonly T[]> | readonly T[] | null,
         matcher: TuiStringMatcher<T> = TUI_DEFAULT_MATCHER,
     ): ReadonlyArray<readonly T[]> | readonly T[] | null {
-        return this.filter(items, matcher, TUI_DEFAULT_STRINGIFY, this.query);
+        return this.filter(
+            items,
+            matcher,
+            this.host.stringify || TUI_DEFAULT_STRINGIFY,
```

Comment:
I don't get it. With custom matcher stringify shouldn't matter because matcher gets the whole item, not its stringified result 🤔

Reply:
Please take a look at [exapmple](https://stackblitz.com/edit/angular-txj8xh-vnvezm?file=src%2Fapp%2Fapp.component.html) (last input) from the #3501

we need to return the whole list if we get a match https://github.com/Tinkoff/taiga-ui/blob/main/projects/kit/pipes/filter-by-input/filter-by-input.base.ts#L47 and we can compare search query and item using actual `stringify` only

Reply:
I marked the pipe as deprecated and removed the example from the demo